pkg/apis/core/v1/defaults.go

```diff
@@ -176,6 +176,36 @@ func SetDefaults_Pod(obj *v1.Pod) {
 				}
 			}
 		}
+		if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+			// set resources allocated equal to requests, if not specified
+			if obj.Spec.Containers[i].Resources.Requests != nil {
+				if obj.Spec.Containers[i].ResourcesAllocated == nil {
```

Comment:
What if the create operation specified ResourcesAllocated?  Then we don't initialize them?

Reply:
In admission plugin, we verify that it matches Resources.Requests, and if not reject the operation.

On the KEP, we had discussed the possibility of user requesting ResourcesAllocated different (typically lower, minimum required) from ideal desired resources in Requests, but agreed to take that up as a future enhancement.

Reply:
As scoped in this KEP, my implementation will fail validation if user specifies ResourcesAllocated that is different from Requests.